From e7e46418bec6f332e33e547ba1bc65476483ff14 Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Mon, 4 Aug 2014 15:37:03 +0800
Subject: [PATCH 16/18] mmc: core: use 10ms delay instead of 1ms

Some of the SD card needs 10ms for power cycle although spec
require 1ms. Change the delay to be 10ms for safe

Change-Id: Ifb604a45e667fbb7918f1f3694ebb51fcb46be77
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/core/core.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 01fd5afe7bcc..a49821f16e5b 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -1675,8 +1675,12 @@ void mmc_power_off(struct mmc_host *host)
 void mmc_power_cycle(struct mmc_host *host, u32 ocr)
 {
 	mmc_power_off(host);
-	/* Wait at least 1 ms according to SD spec */
-	mmc_delay(1);
+	/*
+	 * Wait at least 1 ms according to SD spec
+	 * some of the SD card seems only 1ms is not enough,
+	 * change the actual delay to be 10ms for safe
+	 */
+	mmc_delay(10);
 	mmc_power_up(host, ocr);
 }
 
-- 
1.7.9.5

